### DISCUSSION TITLE
Record over chained pattern

### DISCUSSION THREAD

**Discussion by bukru** [bangcorrupt/hacktribe#301] [Ideas]
First of all thanks to all that allow this project to grow. This is awesome. Shame on korg ^^
I read all the ideas section and i would like to put my suggestion :

For now if you press record button during a chained pattern, it stop chaining pattern so the current pattern is looping. So it's impossible to record nothing while playing chained pattern.

Allow recording while chaining pattern would make a hudge improvement of workflow.

Have a good day : )

**Comment by bangcorrupt** [comment]
This ties in with some changes I want to make to pattern saving.  At the moment, when the pattern changes any unsaved changes are lost.  This is probably why it disables chaining while recording.  I'd like to make it so all changes are kept while switching patterns, then saved permanently to a project file at the end of a session.

**Reply by bukru** [reply]
Maybe it's not possible because writing time... I hope it is.
Does all pattern load into ram at startup ?

**Reply by bangcorrupt** [reply]
Yes, all the patterns are loaded from flash to RAM at boot, then the current pattern is copied to a separate edit buffer.  When we write the pattern, the edit buffer is copied back to the main array in RAM, then written to flash.  It should be possible if we defer writing to non-volatile memory until the sequencer is stopped.

**Reply by scottishcammy** [reply]
Sounds like this might be a fix for the 8-bar restriction? If I have two chained patterns and record over them both,

**Comment by probisilia** [comment]
Something I also wanted to suggest. This only one thing would drastically improve performance

**Reply by javegon** [reply]
Same thoughts here!
